src/TestApp/TcpServer.cs

```diff
@@ -33,19 +30,10 @@ protected override void Receive()
             if (!KeepGoing)
                 return;
 
-            signal.Reset();
-            BoundSocket.BeginAccept(AcceptCallback, BoundSocket);
-            signal.WaitOne();
-        }
-
-        private void AcceptCallback(IAsyncResult asyncResult)
-        {
+            var receivingSocket = BoundSocket.Accept();
```

Comment:
Yes it made an async call, but then made a sync-wait for the async-call to complete without timeout.